### ISSUE TITLE
running tests with ghostscript 9.56 fails

### ISSUE THREAD

**Issue by dirkmueller** [qpdf/qpdf#732]
ghostscript 9.56 has enabled a new pdf interpreter, see https://ghostscript.com/blog/pdfi.html

with that installed, the testsuite is failing, example:

```
gs   -dNOPAUSE -sDEVICE=tiff24nc -sOutputFile=a.tif - < good18.pdf
GPL Ghostscript 9.56.1 (2022-04-04)
Copyright (C) 2022 Artifex Software, Inc.  All rights reserved.
This software is supplied under the GNU AGPLv3 and comes with NO WARRANTY:
see the file COPYING for details.
   **** Error: Couldn't initialise file.
               Output may be incorrect.
   No pages will be processed (FirstPage > LastPage).

The following warnings were encountered at least once while processing this file:
        bad trailer dictionary

   **** This file had errors that were repaired or ignored.
   **** Please notify the author of the software that produced this
   **** file that it does not conform to Adobe's published PDF
```

passing `-dNEWPDF=false` makes it work again for now, however ghostscript is documenting that this switch will be eventually removed.

**Comment by jberkenbilt** [comment]
This appears to be a bug with ghostscript. I reported it here: https://bugs.ghostscript.com/show_bug.cgi?id=705842

**Comment by jberkenbilt** [comment]
I have added a workaround to qpdf's test suite to detect version 9.x, x >= 56 and add the -dNEWPDF=false flag, and a note to myself to adjust the code to work around only broken versions once the bug is fixed.

**Comment by jberkenbilt** [comment]
Got confirmation from ghostscript that the bug is already fixed on their main branch and in the 10.00.0 release candidate, so I'll leave my workaround as 9.x, x >= 56, assuming that, if there is a 9.57, -dNEWPDF=false will still be there.
